src/ProgramRepresentation/ProgramFunction.cpp

```diff
@@ -14,41 +16,43 @@ std::list<ProgramPoint> ProgramFunction::getProgramPoints() {
   return this->programPoints;
 }
 
-ProgramPoint *ProgramFunction::getProgramPointRef(std::string name,
+ProgramPoint *ProgramFunction::getProgramPointRef(std::string pointName,
                                                   bool addNewIfNotFound) {
   for (ProgramPoint &programPoint : this->programPoints) {
-    if (programPoint.getName() == name) {
+    if (programPoint.getPointName() == pointName) {
       return &programPoint;
     }
   }
 
   if (addNewIfNotFound) {
-    ProgramPoint newProgramPoint = ProgramPoint(name);
+    ProgramPoint newProgramPoint = ProgramPoint(pointName);
     this->addProgramPoint(newProgramPoint);
     return &this->programPoints.back();
   }
 
-  return new ProgramPoint();
+  // TODO: program should error and halt if this case happens
+  return NULL;
 }
 
-ProgramPoint ProgramFunction::getProgramPoint(std::string name,
+ProgramPoint ProgramFunction::getProgramPoint(std::string pointName,
                                               bool addNewIfNotFound) {
   for (ProgramPoint &programPoint : this->programPoints) {
-    if (programPoint.getName() == name) {
+    if (programPoint.getPointName() == pointName) {
       return programPoint;
     }
   }
 
   if (addNewIfNotFound) {
-    ProgramPoint newProgramPoint = ProgramPoint(name);
+    ProgramPoint newProgramPoint = ProgramPoint(pointName);
     this->addProgramPoint(newProgramPoint);
     return this->programPoints.back();
   }
 
+  // TODO: program should error and halt if this case happens
```

Comment:
Stopping the program is relatively easy, so let's just do this rather than leaving a TODO and failing silently.